independent-projects/tools/codestarts/src/main/java/io/quarkus/devtools/codestarts/utils/NestedMaps.java

```diff
@@ -57,11 +64,13 @@ public static void deepMerge(Map left, Map right) {
                     || rightValue instanceof Float
                     || rightValue instanceof Long
                     || rightValue instanceof Double
-                    || rightValue instanceof String) {
+                    || rightValue instanceof String
+                    || rightValue == null) {
                 // Override
                 left.put(key, rightValue);
+            } else {
+                throw new IllegalArgumentException("Invalid value type for deepMerge: " + rightValue.getClass());
```

Comment:
@aloubyansky as discussed, it now throws an error if an unexpected object is used as input. I also added a safety to never modify the left map by adding a hat method.